iothub/service/API design.md

```diff
@@ -0,0 +1,88 @@
+```csharp
+
+public class IotHubServiceClient : IDisposable
+{
+	public DevicesClient Devices { get; private set; }
+	public ModulesClient Modules { get; private set; }
+
+	protected IotHubServiceClient();
+	
+	public IotHubServiceClient(string connectionString, IotHubServiceClientOptions options = default);
+	public IotHubServiceClient(string hostName, TokenCredential credential, IotHubServiceClientOptions options = default);
+	public IotHubServiceClient(string hostName, AzureSasCredential credential, IotHubServiceClientOptions options = default);
+
+	public void Dispose();
+}
+
+public class IotHubServiceClientOptions
+{
+	public IotHubServiceClientOptions(ServiceVersion version = LatestVersion)
+
+	public IWebProxy Proxy { get; set; }
+	public TimeSpan HttpConnectionLeaseTimeout { get; set; } = TimeSpan.FromMinutes(5);
```

Comment:
I'm tempted to get rid of this option since the user can provide an HTTP client with this value set on it. Any thoughts?